scripts/o3de/o3de/register.py

```diff
@@ -791,11 +690,101 @@ def register(engine_path: pathlib.Path = None,
+    result = 0
+
+    for project in json_data.get('projects', []):
+        if not validation.valid_o3de_project_json(pathlib.Path(project).resolve() / 'project.json'):
+            logger.warn(f"Project path {project} is invalid.")
+            # Attempt to unregister all invalid projects even if previous projects failed to unregister
+            # but combine the result codes of each command.
+            result = register(project_path=pathlib.Path(project), remove=True) or result
+
+    return result
+
+def remove_invalid_o3de_objects() -> None:
+    for engine_path in manifest.get_manifest_engines():
+        if not validation.valid_o3de_engine_json(pathlib.Path(engine_path).resolve() / 'engine.json'):
+            logger.warn(f"Engine path {engine_path} is invalid.")
+            register(engine_path=engine_path, remove=True)
+
+    remove_invalid_o3de_projects()
+
+    for external in manifest.get_manifest_external_subdirectories():
+        external = pathlib.Path(external).resolve()
+        if not external.is_dir():
+            logger.warn(f"External subdirectory {external} is invalid.")
+            register(engine_path=engine_path, external_subdir_path=external, remove=True)
 
+    for template in manifest.get_manifest_templates():
+        if not validation.valid_o3de_template_json(pathlib.Path(template).resolve() / 'template.json'):
+            logger.warn(f"Template path {template} is invalid.")
+            register(template_path=template, remove=True)
+
+    for restricted in manifest.get_manifest_restricted():
+        if not validation.valid_o3de_restricted_json(pathlib.Path(restricted).resolve() / 'restricted.json'):
+            logger.warn(f"Restricted path {restricted} is invalid.")
+            register(restricted_path=restricted, remove=True)
+
+    json_data = manifest.load_o3de_manifest()
+    default_engines_folder = pathlib.Path(json_data.get('default_engines_folder', manifest.get_o3de_engines_folder())).resolve()
+    if not default_engines_folder.is_dir():
+        new_default_engines_folder = manifest.get_o3de_folder() / 'Engines'
+        new_default_engines_folder.mkdir(parents=True, exist_ok=True)
+        logger.warn(
+            f"Default engines folder {default_engines_folder} is invalid. Set default {new_default_engines_folder}")
+        register(default_engines_folder=new_default_engines_folder.as_posix())
+
+    default_projects_folder = pathlib.Path(json_data.get('default_projects_folder', manifest.get_o3de_projects_folder())).resolve()
+    if not default_projects_folder.is_dir():
+        new_default_projects_folder = manifest.get_o3de_folder() / 'Projects'
+        new_default_projects_folder.mkdir(parents=True, exist_ok=True)
+        logger.warn(
+            f"Default projects folder {default_projects_folder} is invalid. Set default {new_default_projects_folder}")
+        register(default_projects_folder=new_default_projects_folder.as_posix())
```

Comment:
This check is repeated 5 times, its a good candidate to turn into a function call